packages/loader/container-loader/src/container.ts

```diff
@@ -1143,7 +1191,9 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         if (this.service.policies?.caching !== LoaderCachingPolicy.NoCaching) {
             service = new PrefetchDocumentStorageService(service);
         }
-        return service;
+
+        this.retriableStorageService = new RetriableDocumentStorageService(service, this);
```

Comment:
Can we not have this.retriableStorageService as a member variable in this class?
I feel like it would be great to have dispose() on interface, and propagated it through wrappers, as main storage implementation may have things to dispose (like cache in SPO driver). It does not have to be same PR, but I think it would be great change - both in terms of simplifying this layer (not increasing state), but also for driver layers

Reply:
Will follow up on this in follow up PR.